### PR TITLE
Prepare to enable Prettier

### PR DESCRIPTION
**Pull request by octogonz**
This PR adds a `.prettierrc.js` file and does an initial bulk-reformatting of all files to use Prettier.

An upcoming PR will enable the GitHook that keeps everything formatted.

### CODE REVIEW

File: apps/api-documenter/src/cli/ApiDocumenterCommandLine.ts [hunk 1]
```diff
@@ -10,13 +10,15 @@ export class ApiDocumenterCommandLine extends CommandLineParser {
...
+      toolDescription:
+        'Reads *.api.json files produced by api-extractor, ' +
+        ' and generates API documentation in various output formats.',
     });
     this._populateActions();
   }
 
-  protected onDefineParameters(): void { // override
+  protected onDefineParameters(): void {
+    // override
```
Comment (iclanton) [inline apps/api-documenter/src/cli/ApiDocumenterCommandLine.ts:21]: Should this be in a `/** ... */` comment?

File: apps/api-documenter/src/cli/BaseAction.ts [hunk 2]
```diff
@@ -25,22 +22,25 @@ export abstract class BaseAction extends CommandLineAction {
   private _inputFolderParameter: CommandLineStringParameter;
   private _outputFolderParameter: CommandLineStringParameter;
 
-  protected onDefineParameters(): void { // override
+  protected onDefineParameters(): void {
+    // override
```
Comment (iclanton) [inline apps/api-documenter/src/cli/BaseAction.ts:26]: Looks like there are a few of these.

File: apps/api-documenter/src/cli/BaseAction.ts [hunk 3]
```diff
@@ -72,28 +72,33 @@ export abstract class BaseAction extends CommandLineAction {
...
         const inheritDocTag: tsdoc.DocInheritDocTag | undefined = apiItem.tsdocComment.inheritDocTag;
 
         if (inheritDocTag && inheritDocTag.declarationReference) {
           // Attempt to resolve the declaration reference
-          const result: IResolveDeclarationReferenceResult
-            = apiModel.resolveDeclarationReference(inheritDocTag.declarationReference, apiItem);
+          const result: IResolveDeclarationReferenceResult = apiModel.resolveDeclarationReference(
+            inheritDocTag.declarationReference,
+            apiItem
+          );
```
Comment (iclanton) [inline apps/api-documenter/src/cli/BaseAction.ts:84]: This is the parameter style I like. I wish Prettier had an option to always do this if any of the parameters ever wrap.

File: apps/api-documenter/src/documenters/MarkdownDocumenter.ts [hunk 4]
```diff
@@ -4,11 +4,7 @@
 /* eslint max-lines: "off" */
 
 import * as path from 'path';
-import {
-  PackageName,
-  FileSystem,
-  NewlineKind
-} from '@rushstack/node-core-library';
+import { PackageName, FileSystem, NewlineKind } from '@rushstack/node-core-library';
```
Comment (iclanton) [inline apps/api-documenter/src/documenters/MarkdownDocumenter.ts:7]: I do prefer `imports` with multiple imported entities being always wrapped. Too bad there isn't an option for this.

File: apps/api-documenter/src/documenters/MarkdownDocumenter.ts [hunk 5]
```diff
@@ -94,8 +90,8 @@ export class MarkdownDocumenter {
           documenter: new MarkdownDocumenterAccessor({
             getLinkForApiItem: (apiItem: ApiItem) => {
               return this._getLinkFilenameForApiItem(apiItem);
-            }
-          })
+            },
```
Comment (iclanton) [inline apps/api-documenter/src/documenters/MarkdownDocumenter.ts:93]: I thought we didn't want trailing commas.

File: apps/api-documenter/src/documenters/MarkdownDocumenter.ts [hunk 6]
```diff
@@ -57,7 +53,7 @@ import { CustomMarkdownEmitter } from '../markdown/CustomMarkdownEmitter';
 import { PluginLoader } from '../plugin/PluginLoader';
 import {
   IMarkdownDocumenterFeatureOnBeforeWritePageArgs,
-  MarkdownDocumenterFeatureContext
+  MarkdownDocumenterFeatureContext,
```
Comment (iclanton) [inline apps/api-documenter/src/documenters/MarkdownDocumenter.ts:56]: Same here - trailing comma?

Comment (iclanton) [reply apps/api-documenter/src/documenters/MarkdownDocumenter.ts:56]: Looks like there are a bunch of these in this file.

File: webpack/localization-plugin/src/AssetProcessor.ts [hunk 7]
```diff
@@ -225,7 +228,7 @@ export class AssetProcessor {
               } else {
                 issues.push(
                   `The string "${localizedElement.stringName}" in "${localizedElement.locFilePath}" is missing in ` +
-                  `the locale ${locale}`
+                    `the locale ${locale}`
```
Comment (iclanton) [inline webpack/localization-plugin/src/AssetProcessor.ts:231]: This seems inconsistent with other similar changes.

File: apps/api-documenter/src/cli/ApiDocumenterCommandLine.ts (hunk 1 above)
Comment (octogonz) [reply apps/api-documenter/src/cli/ApiDocumenterCommandLine.ts:21]: `// override` and `// abstract` are a convention I use to remember how the base class member is defined. Eventually should replace them with the TSDoc equivalents (and an ESLint rule to valide them) but it's out of scope for this PR that already churns 838 files :-)

File: apps/api-documenter/src/documenters/MarkdownDocumenter.ts (hunk 5 above)
Comment (octogonz) [reply apps/api-documenter/src/documenters/MarkdownDocumenter.ts]: I have updated **.prettierrc.js** based on our conversation

File: apps/api-documenter/src/documenters/MarkdownDocumenter.ts (hunk 4 above)
Comment (octogonz) [reply apps/api-documenter/src/documenters/MarkdownDocumenter.ts:7]: FYI there are two GitHub issues related to this topic, and both of them are asking for the opposite:

https://github.com/prettier/prettier/issues/5253
https://github.com/prettier/prettier/issues/5995

The rationale seems to be "nobody wants to read a huge block of imports, just make it as small as possible".

**Pull request merged**